src/Symfony/Component/DependencyInjection/Tests/Compiler/PriorityTaggedServiceTraitTest.php

```diff
@@ -85,6 +87,38 @@ public function testWithEmptyArray()
         $priorityTaggedServiceTraitImplementation = new PriorityTaggedServiceTraitImplementation();
         $this->assertEquals([], $priorityTaggedServiceTraitImplementation->test('my_custom_tag', $container));
     }
+
+    public function testOnlyTheFirstNonIndexedTagIsListed()
+    {
+        $container = new ContainerBuilder();
+        $container->register('service1')->addTag('my_custom_tag', ['foo' => 'bar']);
+
+        $definition = $container->register('service2', BarTagClass::class);
+        $definition->addTag('my_custom_tag', ['priority' => 100]);
+        $definition->addTag('my_custom_tag', ['foo' => 'a']);
+        $definition->addTag('my_custom_tag', ['foo' => 'b', 'priority' => 100]);
+        $definition->addTag('my_custom_tag', ['foo' => 'b']);
+        $definition->addTag('my_custom_tag', []);
+
+        $priorityTaggedServiceTraitImplementation = new PriorityTaggedServiceTraitImplementation();
+
+        $expected = [
+            new Reference('service2'),
+            new Reference('service1'),
+        ];
+        $this->assertEquals($expected, $priorityTaggedServiceTraitImplementation->test('my_custom_tag', $container));
+
+        $tag = new TaggedIteratorArgument('my_custom_tag', 'foo');
```

Comment:
The case of indexed and non-indexed arguments should be split into separate tests, so that they both run even if the first one fails. It also makes it easier to understand what these tests are covering, making the testsuite easier to maintain.

Reply:
works for me, updated.